### PR TITLE
Add user preference for switching between 12-hour and 24-hour time display

### PR DESCRIPTION
**Pull request by ChaoticWeg**
Also adding the preference screen itself, as part of a new BottomTabNavigator.

### CODE REVIEW

**Comment by ChaoticWeg** [comment]
```
Invariant Violation: withNavigation can only be used on a view hierarchy of a navigator. The wrapped component is unable to get access to navigation from props or context.
```

Tests are failing due to the above, inside `src/screens/__tests__/HomeScreen-test.tsx`. I'm certain this has something to do with the fact that the `HomeScreen` makes use of `<NavigationEvents/>` inside an `<InactivityHandler>`, looking into it now. Might need help, @richardkriesman, when you get the chance.

~~**EDIT:** I'm going to move all of that nonsense over into `HomeScreenClock` and see how it goes.~~

**EDIT 2:** No I'm not, it's a component not a `UIScreen`.

File: src/components/AppSettingsScreen/__tests__/BooleanPreference-test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    env = await TestEnvironment.init();
+    env.timing.date = fakeDateMillis;
+    done();
+});
+
+it("renders properly", () => {
+    const cb = jest.fn();
+
+    const elems = [
+        <BooleanPreference value={true} disabled={false} title="Test 1" onValueChange={cb}/>,
```
Comment (richardkriesman) [inline src/components/AppSettingsScreen/__tests__/BooleanPreference-test.tsx:20]: I love this PreferenceItem stuff :heart_eyes:

**Review by richardkriesman — DISMISSED** [review]
Everything looks good to me! Though I made some changes so could you give it look over before we merge, @ChaoticWeg? I'd assign you as a reviewer but I don't think I can for the PR's owner.

File: src/__tests__/__snapshots__/App-test.tsx.snap [hunk 2]
```diff
@@ -2,391 +2,4 @@
 
 exports[`renders the loading screen 1`] = `null`;
```
Comment (ChaoticWeg) [inline src/__tests__/__snapshots__/App-test.tsx.snap]: App snapshots are broken it seems, both are null.

File: src/components/AppSettingsScreen/TouchPreference.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { ListItemProps } from "react-native-elements";
+import { PreferenceItem } from "./PreferenceItem";
+
+export class TouchPreference extends PreferenceItem<any> {
```
Comment (ChaoticWeg) [inline src/components/AppSettingsScreen/TouchPreference.tsx]: If possible, should make an interface `TouchPreferenceProps extends PreferenceProps<any>` with onPress being required.

**Review by ChaoticWeg — COMMENTED** [review]
Can't approve or request changes, but just had those couple questions. Otherwise, looks good!

**Comment by richardkriesman** [comment]
[Add user preference for switching between 12-hour and 24-hour time display](https://trello.com/c/oPDW8nn1/87-add-user-preference-for-switching-between-12-hour-and-24-hour-time-display)

File: src/__tests__/__snapshots__/App-test.tsx.snap (hunk 2 above)
Comment (richardkriesman) [reply src/__tests__/__snapshots__/App-test.tsx.snap]: Okay, so looking into this, the first one is always null, actually. For the second one, the changes to the database initialization logic prevented anything from rendering. This should be fixed now.

**Review by richardkriesman — APPROVED** [review]

**Pull request merged**